File: literary_life_backend/app/schemas/response.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime


class ResponseCreate(BaseModel):
    work_id: int
    content: str


class ResponseOut(BaseModel):
    id: int
    work_id: int
    user_id: int
    content: str
    created_at: datetime
    author_nickname: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)

File: literary_life_app/lib/models/response.dart
class LiteraryResponse {
  final int id;
  final int workId;
  final int userId;
  final String content;
  final DateTime createdAt;
  final String? authorNickname;

  LiteraryResponse({
    required this.id,
    required this.workId,
    required this.userId,
    required this.content,
    required this.createdAt,
    this.authorNickname,
  });

  factory LiteraryResponse.fromJson(Map<String, dynamic> json) {
    return LiteraryResponse(
      id: json['id'],
      workId: json['work_id'],
      userId: json['user_id'],
      content: json['content'],
      createdAt: DateTime.parse(json['created_at']),
      authorNickname: json['author_nickname'],
    );
  }
}
